### PR TITLE
refactor: replace httplib2 with aiohttp, full async calls

### PR DESCRIPTION
**Pull request by vinteo**
Please help test, including:

- SSL
- Basic Auth

### CODE REVIEW

File: pyopensprinkler/__init__.py [hunk 1]
```diff
@@ -277,15 +272,10 @@ def _get_variables(self):
...
         return self._retrieve_state()["settings"]
 
-    def _set_variable(self, variable, value):
+    async def _set_variable(self, variable, value):
         """Set variable"""
         params = {variable: value}
-        (_, content) = self.request("/cv", params)
-        return content["result"]
-
-    def _set_variables(self, variables):
```
Comment (vinteo) [inline pyopensprinkler/__init__.py:286]: not used

File: pyopensprinkler/__init__.py [hunk 2]
```diff
@@ -255,15 +255,10 @@ def _get_options(self):
...
         return self._retrieve_state()["options"]
 
-    def _set_option(self, option, value):
+    async def _set_option(self, option, value):
         """Set option"""
         params = {option: value}
-        (_, content) = self.request("/co", params)
-        return content["result"]
-
-    def _set_options(self, options):
```
Comment (vinteo) [inline pyopensprinkler/__init__.py:264]: not used

File: pyopensprinkler/program.py [hunk 3]
```diff
@@ -50,48 +50,45 @@ def _set_variables(self, params=None):
...
-        (_, content) = self._controller.request("/mp", params)
+        (_, content) = await self._controller.request("/mp", params)
         return content["result"]
 
     def _get_data_flag_bits(self):
         return list(
             reversed([int(x) for x in list("{0:08b}".format(self._get_variable(0)))])
         )
 
-    def _set_data_flag_bit(self, index, value):
```
Comment (vinteo) [inline pyopensprinkler/program.py:67]: not used

**Comment by travisghansen** [comment]
Looks sane enough. The stuff that has been removed seems unnecessary to remove. Just because they aren't used by the lib directly or even hass integration doesn't mean they aren't (or won't) be used by someone trying to leverage the library for their own projects/purposes.

**Comment by vinteo** [comment]
They were marked as private methods though (starting with _), usually those
will not be used externally

**Comment by vinteo** [comment]
@travisghansen are you able to test the SSL and basic auth? I believe you have those on your setup.

**Comment by travisghansen** [comment]
So make them public instead of removing? Seems like any fleshed out library would have support for those things.

Yeah I can test out ssl/basic auth easily enough. Maybe I can get at it tonight..

**Comment by vinteo** [comment]
I am not sure they would be useful externally. As you know I am on the camp that code maintenance in the long term is way more costly than adding code itself.

File: pyopensprinkler/__init__.py [hunk 4]
```diff
@@ -95,6 +95,7 @@ def __init__(self, url, password, opts=None):
         self._stations = {}
         self._state = None
         self._last_refresh_time = None
+        self._http_client = None
```
[the remark this answers is not in the document]
Comment (vinteo) [reply pyopensprinkler/__init__.py:99]: at session option in `opts` from constructor

File: pyopensprinkler/station.py [hunk 5]
```diff
@@ -69,92 +69,92 @@ def _bit_set(self, bit_property, bit_update_name, value):
...
         """Stop station"""
         params = {"en": 0}
-        return self._manual_run(params)
+        return await self._manual_run(params)
 
-    def toggle(self):
+    async def toggle(self):
         """Toggle station"""
-        self._controller.refresh()
+        await self._controller.refresh()
```
[the remark this answers is not in the document]
Comment (vinteo) [reply pyopensprinkler/station.py:88]: Yea if you put it that way, but I will leave it out of this PR

File: pyopensprinkler/station.py [hunk 6]
```diff
@@ -69,92 +69,92 @@ def _bit_set(self, bit_property, bit_update_name, value):
...
-    def disable(self):
-        return self.set_enabled(False)
+    async def disable(self):
+        return await self.set_enabled(False)
 
-    def set_enabled(self, value):
+    async def set_enabled(self, value):
         bit_property = "stn_dis"
         bit_update_name = "d"
         if value:
```
[the remark this answers is not in the document]
Comment (vinteo) [reply pyopensprinkler/station.py:106]: /cc @travisghansen 

But till do it separately if changes needed

File: pyopensprinkler/station.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (PeteBa) [reply pyopensprinkler/station.py:88]: Resolving in favour of #51

File: pyopensprinkler/station.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (PeteBa) [reply pyopensprinkler/station.py:106]: Resolved in favour of #52

**Comment by vinteo** [comment]
@travisghansen any issues with SSL or auth?

**Comment by vinteo** [comment]
tested SSL and basic auth

**Pull request merged**